packages/gatsby-plugin-catch-links/src/__tests__/catch-links.js

```diff
@@ -340,7 +340,7 @@ describe(`navigation is routed through gatsby if the destination href`, () => {
       view: window,
     })
 
-    hrefHandler.mockImplementation(() => {
+    setTimeout(() => {
```

Comment:
Changed these to `setTimeout` as otherwise there would be a discrepancy between this and the new test, which could have explained the difference in behaviour.

Reply:
I moved to addEvenlistener instead, it makes it more verbose when reading the tests. Feel free to tell me to revert if you think otherwise

Reply:
I think all we need is something that runs after our handler would have been called, either approach will achieve this so I'm happy. I personally think it's fine for tests to be verbose if the verbosity adds meaning, and a click handler vs setTimeout does add meaning.